### PR TITLE
subsys: bootloader: c_runtime_setup: Add more startup files

### PR DESCRIPTION
**Pull request by oyvindronningstad**
For nRF51, nRF52832, and nRF52810.
Also fix CMakeLists.txt.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

### CODE REVIEW

**Comment by oyvindronningstad** [comment]
cc @hakonfam @sigvartmh @carlescufi

**Comment by oyvindronningstad** [comment]
@lemrey @ioannisg Any comments?

**Comment by oyvindronningstad** [comment]
> Is it possible to make the order a bit more logical, i.e 51, 52810, 52832, 52840, 9160?

Good point, done.

**Comment by oyvindronningstad** [comment]
@ioannisg @lemrey Ready to merge

**Comment by lemrey** [comment]
@oyvindronningstad, looks good to me but can we have @sigvartmh or @hakonfam have a look too?

File: subsys/bootloader/fprotect/fprotect_bprot.c [hunk 1]
```diff
@@ -30,7 +30,7 @@ int fprotect_area(u32_t start, size_t length)
 
 	if ((start % BPROT_REGIONS_SIZE) ||
 	    (length % BPROT_REGIONS_SIZE) ||
-	    (pagenum_end >= BPROT_REGIONS_NUM) ||
+	    (pagenum_end > BPROT_REGIONS_NUM) ||
```
Comment (hakonfam) [inline subsys/bootloader/fprotect/fprotect_bprot.c]: Sorry for late review, but shouldn't this be done in a separate commit?

**Comment by oyvindronningstad** [comment]
@carlescufi Merge this?
